That's a good hint (the fact that the native tests have been run with the SNAPSHOTs).

However, this is the detailed log message (irrelevant parts omitted) of the test failure:
```
bb0e-b8ae5a0ddb40-1: org.jboss.resteasy.spi.UnhandledException: io.fabric8.kubernetes.client.KubernetesClientException: io.fabric8.kubernetes.api.model.KubernetesResource: Provider io.fabric8.kubernetes.api.model.gatewayapi.v1alpha2.HTTPRouteList not found
...
2023-06-01T13:43:28.5431445Z Caused by: io.fabric8.kubernetes.client.KubernetesClientException: io.fabric8.kubernetes.api.model.KubernetesResource: Provider io.fabric8.kubernetes.api.model.gatewayapi.v1alpha2.HTTPRouteList not found
2023-06-01T13:43:28.5432300Z 	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.waitForResult(OperationSupport.java:520)
...
2023-06-01T13:43:28.5436671Z 	at io.quarkus.it.kubernetes.client.Pods.deleteFirst(Pods.java:41)
2023-06-01T13:43:28.5444828Z Caused by: java.util.ServiceConfigurationError: io.fabric8.kubernetes.api.model.KubernetesResource: Provider io.fabric8.kubernetes.api.model.gatewayapi.v1alpha2.HTTPRouteList not found
2023-06-01T13:43:28.5445496Z 	at java.base@17.0.7/java.util.ServiceLoader.fail(ServiceLoader.java:593)
...
2023-06-01T13:43:28.5448588Z 	at io.fabric8.kubernetes.internal.KubernetesDeserializer$Mapping.registerClasses(KubernetesDeserializer.java:208)
2023-06-01T13:43:28.5449290Z 	at io.fabric8.kubernetes.internal.KubernetesDeserializer$Mapping.registerClassesFromClassLoaders(KubernetesDeserializer.java:200)
2023-06-01T13:43:28.5449931Z 	at io.fabric8.kubernetes.internal.KubernetesDeserializer.<init>(KubernetesDeserializer.java:88)
2023-06-01T13:43:28.5450643Z 	at io.fabric8.kubernetes.client.utils.KubernetesSerialization.getKubernetesDeserializer(KubernetesSerialization.java:147)
2023-06-01T13:43:28.5451382Z 	at io.fabric8.kubernetes.client.utils.KubernetesSerialization.access$000(KubernetesSerialization.java:64)
2023-06-01T13:43:28.5452193Z 	at io.fabric8.kubernetes.client.utils.KubernetesSerialization$1.deserializerInstance(KubernetesSerialization.java:99)
```

I'm mostly sure it has to do with the more elaborate changed we did to to serialization in this release https://github.com/fabric8io/kubernetes-client/pull/4662

However, Chris' comment makes me believe that maybe is an issue while registering the SPI modules :shrug:

--- a/pom.xml
+++ b/pom.xml
@@ -70,7 +70,7 @@
 
         <!-- Dependency versions -->
         <jacoco.version>0.8.10</jacoco.version>
-        <kubernetes-client.version>6.6.2</kubernetes-client.version> <!-- Please check with Java Operator SDK team before updating -->
+        <kubernetes-client.version>6.7.1</kubernetes-client.version> <!-- Please check with Java Operator SDK team before updating -->
 
         <!-- Make sure to check compatibility between these 2 gRPC components before upgrade -->
         <grpc.version>1.55.1</grpc.version> <!-- when updating, verify if com.google.auth should not be updated too -->

